components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/layout/DefaultSuggestionViewHolder.kt

```diff
@@ -120,29 +116,47 @@ internal sealed class DefaultSuggestionViewHolder {
 internal class IconLoader(
     private val iconView: ImageView
 ) {
-    private val scope = CoroutineScope(Dispatchers.Main)
     @VisibleForTesting
     internal var iconJob: Job? = null
 
     fun load(suggestion: AwesomeBar.Suggestion) {
-        cancel()
-
         val icon = suggestion.icon ?: return
 
-        iconJob = scope.launch {
-            val bitmap = withContext(Dispatchers.IO) {
+        /*
+        Originally we were nulling out the icon and setting it every time.
+        This means that there is a "flicker" since even if the image is the same it must be redrawn.
+        In order to get rid of this flicker, we must make it so a suggestion cannot be served until its icon is ready
+        hence why we're doing a "runBlocking" here instead of launching a scope.
+        This is what desktop does and it seems like a reasonable solution.
```

Comment:
Another option we have here is to just make the `Suggestion` take in a Bitmap directly as an icon instead of a suspend function. This would make it clear to users that they need to pass in an icon that's ready to go. What are our thoughts on this?

Reply:
Passing in bitmap directly solution can be found here: https://github.com/mozilla-mobile/android-components/pull/4655/files